brave/brave-core
# (Wallet) Clean up and improve transaction fragment implementation

## Resolves https://github.com/brave/brave-browser/issues/41122

* Remove unused classes, styles, and methods.
* Perform code refactoring
* Adjust window flags and make status bar translucent
* Improve bottom sheet dialog animation and remove flickering
* Increase bottom sheet dialog height
* Change logic to show transaction fragment without overlapping Wallet panel
* Stylize bottom sheet dialog fragment and introduce rounded corners

## Demo

https://github.com/user-attachments/assets/ac4e2e54-2d55-44a8-bc62-d108619cc867



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
- Navigate to Wallet section and create a transaction
- Observe the Wallet icon in the address bar showing a red dot
- Tap on the Wallet icon and observe the transaction dialog animation is rendered smoothly without flickering
- Dismiss the transaction dialog as you like (swiping or tapping back)
- Observe the Wallet panel pops up when the dialog is dismissed
